src/service.cc

```diff
@@ -408,7 +429,19 @@ Status Service::genotype_sites(const genotyper_config& cfg, const string& sample
             if (ls.bad()) {
                 return ls;
             }
-            results[i] = make_pair(move(bcf),losses_for_site);
+
+            shared_ptr<string> residual_rec = nullptr;
+            if (residuals != nullptr &&
+                any_losses(losses_for_site)) {
+                // create a residuals loss record
+                residual_rec = make_shared<string>();
+                s = residuals->gen_record(sites[i], hdr.get(), bcf.get(), *residual_rec);
```

Comment:
Use `ls` instead of `s` here -- the latter is shared in the enclosing function scope. Lambdas in a mutable environment, what could go wrong?